src/core/load-fragment.c

```
@@ -6573,3 +6575,43 @@ int config_parse_tty_size(
 
         return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
 }
+
+int config_parse_log_filter_regex(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        _cleanup_(pattern_freep) pcre2_code *cs = NULL;
+        ExecContext *c = data;
+        bool allow_list = true;
+        int r;
+
+        assert(c);
+        assert(rvalue);
+
+        if (rvalue[0] == '~') {
+                allow_list = false;
+                ++rvalue;
+        }
+
+        if (isempty(rvalue)) {
+                c->log_filter_regex = mfree(c->log_filter_regex);
+                c->log_filter_allow_list = true;
+                return 0;
+        }
+
+        r = pattern_compile_and_log(rvalue, 0, &cs);
+        if (r < 0)
+                return 0;
```

Still need graceful handling of not supported errors here (log_syntax() and ignore the value instead of failing hard)